### PR TITLE
Add notes/cheat sheet markdown 

### PR DESCRIPTION
**Pull request by vinnyoodles**
Useful for reviewing all the things we've learned the day before an interview/flying to an onsite etc...

### CODE REVIEW

File: NOTES.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+## Notes, Tips + Tricks, and General Study Guide for Interviews
+
+### Table of Contents
+- [Definitions](#definitions)
+- [Tips + Tricks](#tips--tricks)
+
+### Definitions
```
Comment (vinnyoodles) [inline NOTES.md]: @Puyush6889 Can you proofread these definitions and let me know if any of them do not make sense or are confusing?

File: NOTES.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Ex. `foobar => foo`
+
+Ex. `foobar => foba`
+
+Ex. `[ 1, 2, 3, 4, 5 ] => [ 2, 3, 4 ]`
+
+Ex. `[ 1, 2, 3, 4, 5 ] => [ 2, 4 ]`
+
+
+### Tips + Tricks
```
Comment (vinnyoodles) [inline NOTES.md]: @Puyush6889 Feel free to add any tips here that you have come up with that you deem helpful

File: NOTES.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+All sets contain the null element, therefore the null set is always a subset of any set. 
+Because set `b` can only contain elements from `a`, the size of `b` is at most the size of `a`. 
+For a set with `N` elements, there are exactly `2^N` subsets because for every element there are two choices: include the element or exclude the element. 
+This can be visualized as a binary tree when making the recursive call.
+
+Ex. `[ 1, 2, 3, 4, 5 ] => [ 2, 4 ]`
+
+**Permutations**
+> Given a set `a` of elements, a permutation of `a` would be a unique ordering of the elements in `a`. 
+Because the ordering matters for this, the length of a permutation is always equal to the length of the original set.
```
Comment (vinnyoodles) [inline NOTES.md]: @Puyush6889 if you want to wrap text in bold, then surround the text with two asterisks (with no spaces)

```
** this is bold **  (with spaces)
**this is bold** (without spaces)
```

File: NOTES.md (hunk 2 above)
Comment (Puyush6889) [inline NOTES.md]: ok 👍

**Pull request merged**